edge-modules/api-proxy-module/src/monitors/config_parser.rs

```diff
@@ -93,19 +94,127 @@ impl ConfigParser {
+          Err(e) => { 
+              error!("{}", e); 
+              expression 
+            }
+     }    
+}
+
+// Gomma in env var name is forbidden because of that
+// We use this instead of simpler regex because we cannot match overlapping regular expression with regex.
+fn replace_env_var_with_boolean(expression: &str) -> String {
+    let mut fifo: Vec<char> = Vec::new();
+    let mut flush_fifo: Vec<char> = Vec::new();
+
+    for c in expression.chars() {
+        match c {
+            '(' | '!' | '&' | '|' => fifo.push(c),
+            ')' | ',' => {
+                if flush_fifo.len() == 1 && Some('0') == flush_fifo.pop() {
+                    fifo.push('0')
+                } else if flush_fifo.len() > 1 {
+                    fifo.push('1')
+                };
+                fifo.push(c);
+                flush_fifo.clear();
+            }
+            _ => flush_fifo.push(c),
+        }
+    }
+
+    fifo.iter().collect()
+}
+
+// The resolution of the regular expression is done by using a stack.
+// For example: &(!(0),1,1)
+// Stack fills up: Stack = ['&', '(', '!', '(','0']
+// When ")" is encounter, the deepest boolean expression is solved:
+// Stack result Stack = ['&', '(', '1']
+// Stack fills up: Stack = ['&', '(', '1','0','0']
+// When ")" is encounter, the last boolean expression is solved:
+// First all the value are load in temporary stack: tmp_fifo = ['1','0','0']
+// Then the operator '&' is extracted and the expression is solved
+fn solve_boolean_expression(expression: &str) -> Result<String, Error> {
+    let mut fifo: Vec<Expr> = Vec::new();
+
+    for c in expression.chars() {
+        match c {
+            '(' | ',' | ' ' => (),
+            '!' | '|' | '&' => fifo.push(Expr::Operator(c)),
+            '1' => fifo.push(Expr::Value(true)),
+            '0' => fifo.push(Expr::Value(false)),
+            ')' => {
+                let mut tmp_fifo: Vec<bool> = vec![];
+                while let Some(Expr::Value(v)) = fifo.last() {
+                    tmp_fifo.push(*v);
+                    fifo.pop();
+                }
+
+                if let Some(Expr::Operator(val)) = fifo.pop() {
+                    let result = match val {
+                        '!' => {
+                            if tmp_fifo.len() > 1 {
+                                return Err(anyhow::anyhow!("Invalid ! expression {}", expression));
+                            }
+                            !tmp_fifo[0]
+                        }
+                        '|' => tmp_fifo.iter().any(|v| *v),
+                        '&' => tmp_fifo.iter().all(|v| *v),
+                        _ => {
+                            return Err(anyhow::anyhow!("Invalid boolean expression {}", expression));
+                        }
+                    };
+                    fifo.push(Expr::Value(result));
+                } else {
+                    return Err(anyhow::anyhow!("Could not find operator for boolean expression {}", expression));
```

Comment:
```suggestion
                    return Err(anyhow!("Could not find operator for boolean expression {}", expression));
```
usually we import macroses